### ISSUE TITLE
Invalid "category" parameter when reloading the categories page 

### ISSUE THREAD

**Issue by ioanarusiczki** [mozilla/addons-frontend#8163]
STR:
1. Load AMO dev with an android device and go to Extensions tab
2. Tap "Explore all categories"
3. After the categories are displayed reload the page

Expected result:
Page is displayed with no errors

Actual result:
"Invalid "category" parameter. " is displayed 

Notes:
If I paste the link in new tab on desktop for
https://addons.allizom.org/en-US/android/extensions/categories/
https://addons.allizom.org/en-US/android/themes/categories/ 
https://addons.allizom.org/en-US/firefox/extensions/categories/
https://addons.allizom.org/en-US/firefox/themes/categories/ 
I automatically receive the error. There's a 301Moved Permanently in the Network.
The links try to redirect to a search page such as https://addons.allizom.org/en-US/firefox/search/?category=categories&sort=recommended%2Cusers&type=statictheme
`category=categories` is an invalid parameter. 

<img width="645" alt="categories page" src="https://user-images.githubusercontent.com/33448286/59363053-3bf5d980-8d3d-11e9-9138-d13599b9d6ac.png">

This is reproducible on all AMO servers.

**Comment by willdurand** [comment]
I left a comment in a related issue: https://github.com/mozilla/addons-frontend/issues/8123#issuecomment-501321551

TL;DR: the "all categories" URLs look like a category/theme URL and the nginx rules catch them, which explains why users get redirected to the search results with an invalid category (because `categories` isn't a valid category).

All categories URLs:

- addons.allizom.org/en-US/android/extensions/categories/
- addons.allizom.org/en-US/android/themes/categories/
- addons.allizom.org/en-US/firefox/extensions/categories/
- addons.allizom.org/en-US/firefox/themes/categories/

A random category URL:

- addons.mozilla.org/en-US/firefox/extensions/tabs/

**Comment by bobsilverberg** [comment]
What this means is that we need the nginx rules changed so that none of:

- addons.allizom.org/en-US/android/extensions/categories/
- addons.allizom.org/en-US/android/themes/categories/
- addons.allizom.org/en-US/firefox/extensions/categories/
- addons.allizom.org/en-US/firefox/themes/categories/

redirect, but any url with a category other than `categories` does continue to redirect.

These rules are slated to land on prod with tomorrow's push. @bqbn Do you think we can get this new change in before then?

**Comment by bqbn** [comment]
The updated Nginx rules is deployed to -dev and -stage. Please try again.

(may need to clear browser cache before testing)

**Comment by bobsilverberg** [comment]
> The updated Nginx rules is deployed to -dev and -stage. Please try again.
> 
> (may need to clear browser cache before testing)

Looks like it works, thanks for the quick fix @bqbn!

**Comment by ioanarusiczki** [comment]
@bobsilverberg @willdurand 

It only works right now on AMO dev for 
https://addons-dev.allizom.org/en-US/firefox/extensions/categories/
https://addons-dev.allizom.org/en-US/firefox/themes/categories/
but if I change `firefox` to `android` these links still redirect to search results
https://addons-dev.allizom.org/en-US/android/extensions/categories/
https://addons-dev.allizom.org/en-US/android/themes/categories/

On AMO stage all of them redirect to search. 

As  I said in the original [issue](https://github.com/mozilla/addons-frontend/issues/8163#issue-455270821) I can reproduce it on prod. 
Click on https://addons.mozilla.org/en-US/android/extensions/categories/

This is why I don't think it's a problem that appeared this week caused by new nginx rules https://github.com/mozilla/addons-frontend/issues/8123

The weird thing is that if I go to the categories pages on a phone the categories are displayed and I can choose one and that's also displayed (I added it in the STR above) , but once I reload the category page it throws this error (redirects to search results).

**Comment by ioanarusiczki** [comment]
@bobsilverberg @willdurand @bqbn 

I tested it again on AMO dev and AMO stage  - FF67(Win10 and android 8.0) 
The category pages will redirect as expected to the list of categories.
It will be validated later today on prod too. 

![works for me](https://user-images.githubusercontent.com/33448286/59418882-78bfe000-8dd2-11e9-9858-8670b64eacc1.gif)

**Comment by ioanarusiczki** [comment]
Verified on AMO prod - FF67 - Win10
